webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.js

```diff
@@ -0,0 +1,65 @@
+import React, { createContext } from 'react';
+import { Modal } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import Header from './subcomponents/ForemanModalHeader';
+import Footer from './subcomponents/ForemanModalFooter';
+// import { translate as __ } from '../../common/I18n';
+
+export const ModalContext = createContext(null);
+
+const ForemanModal = props => {
+  const { isOpen, onClose, title } = props;
+  if (!isOpen) return null;
+  // Extract header and footer from children, if provided
+  let children = React.Children.toArray(props.children);
+  const headerChild = children.find(
+    child => child.type === ForemanModal.Header
+  );
+  const footerChild = children.find(
+    child => child.type === ForemanModal.Footer
+  );
+  children = children.filter(
+    child => child !== headerChild && child !== footerChild
+  );
+  // Use context to provide values to all descendants (not just direct children)
+  const context = {
+    isOpen,
+    onClose,
+    title,
+  };
+  // Change the name of the props we are passing down to Modal to conform to Patternfly 3 api
+  const newProps = Object.assign({}, props);
+  delete newProps.onClose;
+  delete newProps.isOpen;
+
+  return (
+    <ModalContext.Provider value={context}>
+      <aside className="foreman-modal-content" style={{ padding: '1em' }}>
+        <Modal onHide={onClose} show={isOpen} {...newProps}>
```

Comment:
please use lodash `omit` or patternfly's `excludeKeys` in order of removing props:
```<Modal onHide={onClose} show={isOpen} {...omit(props, ['isOpen','onClose'])}>```
this would also spare the `newProps` copy and deletion of props above.